dpgen/data/gen.py

```diff
@@ -237,12 +237,24 @@
 def poscar_scale(poscar_in, poscar_out, scale):
     with open(poscar_in) as fin:
         lines = list(fin)
-    if "D" == lines[7][0] or "d" == lines[7][0]:
+
+    # Determine if "Selective dynamics" is present
+    if lines[7].strip().lower().startswith("s"):
+        coord_type_line = 8  # If present, coordinates type is on line 9
+    else:
+        coord_type_line = 7  # If not, coordinates type is on line 8
```

Comment:
**Suggestion:** Use the `str.lower()` method to convert the string to lowercase before checking if it starts with a specific character. This will make the code more readable and maintainable. [readability]
```suggestion
    if lines[7].strip().lower().startswith("s"):
        coord_type_line = 8
    else:
        coord_type_line = 7
```